FishBun/src/main/java/com/sangcomz/fishbun/adapter/view/AlbumListAdapter.kt

```diff
@@ -0,0 +1,57 @@
+package com.sangcomz.fishbun.adapter.view
+
+import android.app.Activity
+import android.content.Intent
+import android.net.Uri
+import androidx.recyclerview.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.LinearLayout
+
+import com.sangcomz.fishbun.Fishton
+import com.sangcomz.fishbun.R
+import com.sangcomz.fishbun.bean.Album
+import com.sangcomz.fishbun.define.Define
+import com.sangcomz.fishbun.ui.picker.PickerActivity
+import kotlinx.android.synthetic.main.album_item.view.*
+
+class AlbumListAdapter : RecyclerView.Adapter<AlbumListAdapter.ViewHolder>() {
+    private val fishton = Fishton.getInstance()
+    var albumList = emptyList<Album>()
+        set(value) {
+            field = value
+        }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.album_item, parent, false)
+        return ViewHolder(view, fishton.albumThumbnailSize)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        val uri: Uri = Uri.parse(albumList[position].thumbnailPath)
+        holder.itemView.img_album_thumb.apply {
```

Comment:
I know there are some caveats when using the Kotlin Android Extension. When using this in an activity, cache it so that you can call findViewById only once. However, when used in an Adapter, it can't cache and repeatedly calls findViewById.
   
Please check this.

Reply:
It can be helped from [this site](https://www.androidhuman.com/lecture/kotlin/2017/11/26/kotlin_android_extensions_on_viewholder/)

Reply:
I have changed the ViewHolder class to use LayoutContainer in android extensions library.
And findViewById is not called in onBindViewHolder.

But I'm not sure that fishbun is able to use LayoutContainer.
Because it is the experimental feature in Android now.
What do you think?

Reply:
I don't think it's a good idea to use the LayoutContainer.

I think it's a good idea to prevent findViewById from being called through this method in the link.
```java
class RepositoryHolder(parent: ViewGroup) : RecyclerView.ViewHolder(
        LayoutInflater.from(parent.context)
                .inflate(R.layout.item_repository, parent, false)) {

    // 각 뷰의 인스턴스를 저장하는 프로퍼티를 추가합니다.
    // 생성자가 호출되는 시점에 뷰의 인스턴스가 할당됩니다.
    val ivProfile = itemView.ivItemRepositoryProfile

    val tvName = itemView.tvItemRepositoryName

    val tvLanguage = itemView.tvItemRepositoryLanguage
}
```

Reply:
I changed ViewHolder class.

~~~Kotlin
    class ViewHolder(parent: ViewGroup, albumSize: Int) : RecyclerView.ViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.album_item, parent, false)) {
        val imgALbumThumb = itemView.img_album_thumb
        val txtAlbumName = itemView.txt_album_name
        val txtAlbumCount = itemView.txt_album_count

        init {
            imgALbumThumb.layoutParams = LinearLayout.LayoutParams(albumSize, albumSize)
        }
    }
~~~